### PR TITLE
Kernel.++/2: comment on improper lists being returned

### PR DESCRIPTION
**Pull request by eksperimental**
There is one thing that could be improved, and it is the summary of the function:

> Concatenates two lists.

where right operand does not necessarily need to be a list

### CODE REVIEW

File: lib/elixir/lib/kernel.ex [hunk 1]
```diff
@@ -942,6 +943,13 @@ defmodule Kernel do
       iex> 'foo' ++ 'bar'
       'foobar'
 
+      # returns an improper list
+      iex> [1] ++ 2
+      [1 | 2]
+
+      # returns a proper list
+      iex> [1] ++ [2]
+      [1, 2]
```
Comment (whatyouhide) [inline lib/elixir/lib/kernel.ex]: Do we need to show this case as well? It's very similar to the first case. :\

Comment (eksperimental) [inline lib/elixir/lib/kernel.ex]: well, yes, but i just want to show how it should be written to return a proper list for the same case as when an improper list is returned

**Comment by eksperimental** [comment]
please guys also give me your opinion on the summary question

**Comment by whatyouhide** [comment]
@eksperimental maybe we can say

> Concatenates a list and a term, returning a list.

And then we explain proper/improper lists below. Wdyt?

**Comment by eksperimental** [comment]
@whatyouhide i think that is SPOT ON
let's wait for somebody else's comment so I can rebase this

**Comment by josevalim** [comment]
That looks good!

-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by eksperimental** [comment]
I have added a second commit that adds information about the special case of dealing with improper lists. please have a look and tell me what you guys think.

File: lib/elixir/lib/kernel.ex [hunk 2]
```diff
@@ -942,6 +944,21 @@ defmodule Kernel do
...
+
+      # not a proper list on the left will raises an error
+      iex> [1 | 2] ++ 3
+      ** (ArgumentError) argument error
+
+      # improper list on the right will return an improper list
+      iex> [1] ++ [2 | 3]
+      [1, 2 | 3]
```
Comment (whatyouhide) [inline lib/elixir/lib/kernel.ex]: We usually leave an empty line after code blocks before the closing `"""` :)

Comment (eksperimental) [inline lib/elixir/lib/kernel.ex]: fixed

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**